samples/Azure.Management.Storage/Generated/Models/BlobContainerCollectionGetAllOptions.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.Management.Storage.Models
+{
+    /// <summary> The BlobContainerCollectionGetAllOptions. </summary>
+    public partial class BlobContainerCollectionGetAllOptions
+    {
+        /// <summary> Initializes a new instance of BlobContainerCollectionGetAllOptions. </summary>
+        public BlobContainerCollectionGetAllOptions()
+        {
+        }
+
+        /// <summary> Optional. Specified maximum number of containers that can be included in the list. </summary>
+        public int? Maxpagesize { get; set; }
```

Comment:
do we have any cases where the original default on the method was not the type default?  Meaning do we ever have `int foo = 5`?

If so should we create the options bag such that it explicitly honors all the defaults?

`public int? Maxpagesize {get; set;} = null;`

This way we are covered when the default case is used?

Reply:
Yes, I agree that we should explicitly honor the default value, now all the optional parameters will be generated like [this](https://github.com/Azure/autorest.csharp/blob/d76abeb8c49264eef9de1c8c6dd05f01b9c25f2b/test/TestProjects/MgmtPropertyBag/Generated/Models/FooCollectionGetAllOptions.cs#L21-L24).